File: THOK/Fusion创建新模块所需的文件列表和文件/src/Domain/Fusion.Context.Warehouse.Domain/Models/Bill/ManualMaterialBills/ManualMaterialBillDetail.cs
using System;
using System.Linq;

/*
* 2020-07-31 Create by wuzuqiang
*/
namespace Fusion.Context.Warehouse.Domain.Models.Bill.ManualMaterialBills
{
    public class ManualMaterialBillDetail
    {
        public ManualMaterialBillDetail() { }

        public ManualMaterialBillDetail(
            string brandCode, string brandName,
            string machineCode, string machineName,
            string teamCode, string teamName,
            string originWorkNo,
            string shiftCode = "", string shiftName="")
        {
            this.Id = Guid.NewGuid();
            this.BrandCode = brandCode;
            this.BrandName = brandName;
            this.MachineCode = machineCode;
            this.MachineName = machineName;
            this.TeamCode = teamCode;
            this.TeamName = teamName;
            this.OriginWorkNo = originWorkNo;
            this.ShiftCode = shiftCode;
            this.ShiftName = shiftName;
        }

        public ManualMaterialBillDetail(
            string manualMaterialBillNo,
            int serialNumber,
            string teamCode,
            string teamName,
            string machineCode,
            string machineName,
            string brandCode,
            string brandName)
        {
            this.Id = Guid.NewGuid();
            this.ManualMaterialBillNo = manualMaterialBillNo;
            this.SerialNumber = serialNumber;
            this.TeamCode = teamCode;
            this.TeamName = teamName;
            this.MachineCode = machineCode;
            this.MachineName = machineName;
            this.BrandCode = brandCode;
            this.BrandName = brandName;
            this.ManualMaterialBillDetailStatus = 0;
            this.CreateTime = DateTime.Now;
            this.UpdateTime = DateTime.Now;
        }

        public Guid Id { get; private set; }
        public int SerialNumber { get; private set; }
        public string ManualMaterialBillNo { get; private set; }
        public string TeamCode { get; private set; }
        public string TeamName { get; private set; }
        /// <summary>
        /// 班次，1:夜,2:白,3:中
        /// </summary>
        public string ShiftCode { get; private set; }
        public string ShiftName { get; private set; }
        public string MachineCode { get; private set; }
        public string MachineName { get; private set; }
        public string BrandCode { get; private set; }
        public string BrandName { get; private set; }
        public string OriginWorkNo { get; private set; }
        public ManualMaterialBillDetailStatus ManualMaterialBillDetailStatus { get; set; }
        public DateTime? ProductionBeginTime { get; set; }
        public DateTime? ProductionEndTime { get; set; }
        public DateTime CreateTime { get; set; }
        public DateTime UpdateTime { get; set; }
        public byte[] RowVersion { get; set; }

        public virtual ManualMaterialBill ManualMaterialBill { get; private set; }

        public void Modify(ManualMaterialBillDetail manualMaterialBillDetail)
        {
            if (this.RowVersion.Except(manualMaterialBillDetail.RowVersion).Count() > 0)
            {
                throw new Exception("记录已被修改，请刷新！");
            }

            this.TeamCode = manualMaterialBillDetail.TeamCode;
            this.TeamName = manualMaterialBillDetail.TeamName;
            this.MachineCode = manualMaterialBillDetail.MachineCode;
            this.MachineName = manualMaterialBillDetail.MachineName;
            this.BrandCode = manualMaterialBillDetail.BrandCode;
            this.BrandName = manualMaterialBillDetail.BrandName;
            this.OriginWorkNo = manualMaterialBillDetail.OriginWorkNo;
            this.ManualMaterialBillDetailStatus = manualMaterialBillDetail.ManualMaterialBillDetailStatus;
        }

        public void ModifyManualMaterialBillDetailStatus(ManualMaterialBillDetailStatus status, DateTime time)
        {
            switch (Convert.ToInt32(status))
            {
                case 1:
                    this.ProductionBeginTime = time;
                    break;
                case 2:
                    this.ProductionEndTime = time;
                    break;
            }
            this.ManualMaterialBillDetailStatus = status;
        }

        public void Modify(ManualMaterialBillDetailStatus ManualMaterialBillDetailStatus)
        {
            this.ManualMaterialBillDetailStatus = ManualMaterialBillDetailStatus;
        }

        public bool IsCompleted()
        {
            return this.ManualMaterialBillDetailStatus == ManualMaterialBillDetailStatus.Complete;
        }

        public bool IsChangeShift()
        {
            return this.ManualMaterialBillDetailStatus == ManualMaterialBillDetailStatus.ChangeShift;
        }

        public bool IsChangeBrand()
        {
            return this.ManualMaterialBillDetailStatus == ManualMaterialBillDetailStatus.ChangeBrand;
        }

        internal void TryProduction()
        {
            if (this.ManualMaterialBillDetailStatus != ManualMaterialBillDetailStatus.Wait)
            {
                return;
            }

            this.ManualMaterialBillDetailStatus = ManualMaterialBillDetailStatus.Production;
            this.ProductionBeginTime = DateTime.Now;
        }

        internal void Complete()
        {
            this.ManualMaterialBillDetailStatus = ManualMaterialBillDetailStatus.Complete;
            this.ProductionEndTime = DateTime.Now;
        }

        internal void SetSerialNumber(int serialNumber)
        {
            this.SerialNumber = serialNumber;
        }
        internal void Production()
        {
            this.ManualMaterialBillDetailStatus = ManualMaterialBillDetailStatus.Production;
        }

        internal void ChangeShift()
        {
            this.ManualMaterialBillDetailStatus = ManualMaterialBillDetailStatus.ChangeShift;
        }

        internal void ChangeBrand()
        {
            this.ManualMaterialBillDetailStatus = ManualMaterialBillDetailStatus.ChangeBrand;
        }
    }
}

File: THOK/Fusion创建新模块所需的文件列表和文件/src/User Interface/Fusion/Extjs/Fusion/classic/src/Warehouse/view/Bill/ManualMaterialBills/ManualMaterialBillDetail/ManualMaterialBillDetail.js
/*
* 
*/
Ext.define('Fusion.warehouse.view.bill.ManualMaterialBills.ManualMaterialBillDetail.ManualMaterialBillDetail', {
    extend: 'Ext.grid.Panel',
    alias: 'widget.manualMaterialBillDetail',

    requires: [
        'Fusion.warehouse.view.bill.ManualMaterialBills.ManualMaterialBillDetail.ManualMaterialBillDetailController',
        'Fusion.warehouse.view.bill.ManualMaterialBills.ManualMaterialBillDetail.ManualMaterialBillDetailToolbar',
        'Fusion.warehouse.store.bill.ManualMaterialBills.ManualMaterialBillDetailStore'
    ],
    uses: [
        'Fusion.warehouse.view.bill.ManualMaterialBills.ManualMaterialBillDetail.FindManualMaterialBillDetail',
        'Fusion.warehouse.view.bill.ManualMaterialBills.ManualMaterialBillDetail.ModifyManualMaterialBillDetail'
    ],

    moduleId: null,
    controller: 'manualMaterialBillDetail',

    serialNo: '流水号',
    manualMaterialBillNo: '计划单号',
    teamCode: '班组编码',
    teamName: '班组',
    machineCode: '机组编码',
    machineName: '机组',
    brandCode: '牌号编码',
    brandName: '牌号',
    syncMoveBillNo: '同步移库单',
    status: '单据状态',
    startTime: '作业开始时间',
    finishTime: '作业结束时间',
    originWorkNo: '来源工单号',
    createTime: '创建时间',
    updateTime: '更新时间',

    config: {
        manualMaterialBillNo: null,
        manualMaterialBill: null,
        id: null
    },

    multiColumnSort: true,
    selType: "checkboxmodel",
    selModel: {
        injectCheckbox: 1,
        mode: "MULTI",
        checkOnly: false
    },

    plugins: [
        'gridfilters',
        'gridexporter'
    ],

    listeners: {
        selectionChange: 'onSelectionChange'
    },

    initComponent: function () {
        var me = this,
            record;

        me.manualMaterialBillDetailStatusStore = Ext.create('Fusion.CommonStore.setting.DictionaryDetailStore', {
            storeId: 'manualMaterialBillDetailStatusStore',
            proxy: { extraParams: { dictionaryKey: 'ManualMaterialBillDetailStatus' } }
        });

        me.columns = [
            { xtype: "rownumberer", align: 'center' },
            { header: 'Id', dataIndex: 'Id', hidden: true },
            { header: me.serialNo, dataIndex: 'SerialNumber', width: 100, filter: 'number' },
            { header: me.manualMaterialBillNo, dataIndex: 'ManualMaterialBillNo', hidden: true, filter: 'string', width: 100 },
            { header: me.teamCode, dataIndex: 'TeamCode', filter: 'string', hidden: true, width: 100 },
            { header: me.teamName, dataIndex: 'TeamName', filter: 'string', width: 150 },
            { header: me.machineCode, dataIndex: 'MachineCode', hidden: true, filter: 'string', width: 100 },
            { header: me.machineName, dataIndex: 'MachineName', quickSearch: true, width: 150, filter: 'string' },
            { header: me.brandCode, dataIndex: 'BrandCode', hidden: true, filter: 'string', width: 100 },
            { header: me.brandName, dataIndex: 'BrandName', quickSearch: true, width: 150, filter: 'string' },
            { header: me.syncMoveBillNo, dataIndex: 'SyncMoveBillNo', hidden: true, filter: 'string', width: 100 },
            {
                header: me.status, dataIndex: 'ManualMaterialBillDetailStatus', width: 100,
                filter: {
                    type: 'list',
                    store: me.manualMaterialBillDetailStatusStore,
                    idField: 'Value',
                    labelField: 'Name'
                },
                renderer: function (value) {
                    record = me.manualMaterialBillDetailStatusStore.findRecord('Value', value);
                    return record == null ? value : record.data.Name;
                }
            },
            { header: me.startTime, dataIndex: 'ProductionBeginTime', width: 130, xtype: 'datecolumn', format: 'Y/m/d H:i:s', filter: 'date' },
            { header: me.finishTime, dataIndex: 'ProductionEndTime', width: 130, xtype: 'datecolumn', format: 'Y/m/d H:i:s', filter: 'date' },
            { header: me.originWorkNo, dataIndex: 'OriginWorkNo', width: 200, filter: 'string' },
            { header: me.createTime, dataIndex: 'CreateTime', xtype: 'datecolumn', width: 130, hidden: true, format: 'Y/m/d H:i:s', filter: 'date' }, 
            { header: me.updateTime, dataIndex: 'UpdateTime', xtype: 'datecolumn', width: 130, hidden: true, format: 'Y/m/d H:i:s', filter: 'date' }
        ];

        me.store = Ext.create('Fusion.warehouse.store.bill.ManualMaterialBills.ManualMaterialBillDetailStore');

        me.dockedItems = [{
            xtype: 'manualMaterialBillDetailToolbar',
            dock: 'top',
            moduleId: me.moduleId,
            store: me.store,
            columns: me.columns
        }, {
            xtype: 'pagingtoolbar',
            dock: 'bottom',
            displayInfo: true,
            store: me.store
        }];

        me.callParent();
    },

    destroy: function () {
        this.manualMaterialBillDetailStatusStore = Ext.destroy(this.manualMaterialBillDetailStatusStore);

        this.callParent();
    }
});